db/schema_tables.cc

```diff
@@ -613,14 +616,28 @@ schema_ptr aggregates() {
     }
 #endif
 
+static
+mutation
+redact_columns_for_missing_features(mutation m, schema_features features) {
+    if (features.contains(schema_feature::CDC_OPTIONS)) {
+        return std::move(m);
+    }
+    if (m.schema()->cf_name() != SCYLLA_TABLES) {
+        return std::move(m);
+    }
+    slogger.debug("adjusting schema_tables mutation due to possible in-progress cluster upgrade");
+    m.upgrade(scylla_tables(features));
```

Comment:
We should just remove the `cdc` column here, instead of passing `features` to `scylla_tables()`.

Reply:
This is removing the cdc column, no? And adjusting the mutation's schema.

Reply:
Yes, it does. What I don't like is that it pushes this removal down to `schema_tables()`. If we keep using this pattern, `schema_tables()` will be full of branches that conditionally add columns to the schema, when this code is used only in a brief transitional period. I think we should strive to contain such redacting code to an as small as possible area, instead of spreading it in the codebase. In theory all such redacting will be removed one day and it will be much easier if this will just entail removing one function, as opposed to going through the code and tracing all the different places this logic have spread to.

Reply:
schema_features is a way to handle rolling upgrade in other places so I think we should be consistent and have a single approach.

Reply:
Notice that there's no if in scylla_tables. It selects schema from a table based on condition.

Reply:
I see this:
```
        if (has_cdc_options) {
            sb.with_column("cdc", map_type_impl::get_instance(utf8_type, utf8_type, false));
        }
```

Reply:
it's executed only once:

```
schema_ptr scylla_tables(schema_features features) {
    static auto make = [] (bool has_cdc_options) -> schema_ptr {
        auto id = generate_legacy_id(NAME, SCYLLA_TABLES);
        auto sb = schema_builder(NAME, SCYLLA_TABLES, std::make_optional(id))
            .with_column("keyspace_name", utf8_type, column_kind::partition_key)
            .with_column("table_name", utf8_type, column_kind::clustering_key)
            .with_column("version", uuid_type)
            .set_gc_grace_seconds(schema_gc_grace)
            .with_version(generate_schema_version(id));
        if (has_cdc_options) {
            sb.with_column("cdc", map_type_impl::get_instance(utf8_type, utf8_type, false));
        }
        return sb.build();
    };
    +---- notice the static
    v
    STATIC thread_local schema_ptr schemas[2] = { make(false), make(true) };
    return schemas[features.contains(schema_feature::CDC_OPTIONS)];
}
```

Reply:
Ok, maybe I'm overreacting this. I would have done it something like:
```
auto adjusted_s = schema_builder(schema_tables()).remove_column("cdc_options").build();
m.upgrade(adjusted_s);
```
Just to keep all this adjusting/redacting thing localized to one function. I won't insist though.

Reply:
This breaks down when you start having multiple orthogonal features. Each feature is tied to a column, so if you have many features, you end up with many possible variations.